### PR TITLE
Add support for custom HTTP headers in API calls 

### PR DESCRIPTION
**Pull request by sveisvei**

### CODE REVIEW

File: test/metrics.test.js [hunk 1]
```diff
@@ -64,6 +64,36 @@ test.cb('should sendMetrics and register when metricsInterval is a positive numb
...
+        headers: {
+            randomKey,
+        },
+    });
+
+    metrics.count('toggle-x', true);
+    metrics.count('toggle-x', false);
+    metrics.count('toggle-y', true);
+
+    metrics.on('registered', () => {
```
Comment (SimenB) [inline test/metrics.test.js]: `once`?

File: src/repository.ts [hunk 2]
```diff
@@ -98,25 +104,30 @@ export default class Repository extends EventEmitter implements EventEmitter {
...
+                        {} as Object
+                    );
                     this.storage.reset(obj);
                     this.etag = res.headers.etag;
                     this.emit('data');
                 } catch (err) {
                     this.emit('error', err);
                 }
-            },
+            }
```
Comment (SimenB) [inline src/repository.ts]: huh

File: test/metrics.test.js (hunk 1 above)
Comment (sveisvei) [inline test/metrics.test.js]: Sure, can remove the handler here as well.

File: test/metrics.test.js [hunk 3]
```diff
@@ -64,6 +64,33 @@ test.cb('should sendMetrics and register when metricsInterval is a positive numb
...
+
+    metrics.count('toggle-x', true);
+    metrics.count('toggle-x', false);
+    metrics.count('toggle-y', true);
+
+    metrics.on('sent', () => {
+        t.true(regEP.isDone());
+        t.true(metricsEP.isDone());
+        t.end();
+        metrics.stop();
```
Comment (SimenB) [inline test/metrics.test.js]: Stop before end?

**Pull request merged**